rust/otap-dataflow/crates/otap/src/otap_exporter.rs

```diff
@@ -152,14 +161,25 @@ impl local::Exporter<OtapPdata> for OTAPExporter {
                 // handle control messages
                 Message::Control(NodeControlMsg::TimerTick { .. })
                 | Message::Control(NodeControlMsg::Config { .. }) => {}
+                Message::Control(NodeControlMsg::CollectTelemetry {
+                    mut metrics_reporter,
+                }) => {
+                    _ = metrics_reporter.report(&mut self.pdata_metrics);
+                }
                 // shutdown the exporter
                 Message::Control(NodeControlMsg::Shutdown { .. }) => {
                     // ToDo: add proper deadline function
                     break;
                 }
                 //send data
-                Message::PData(message) => {
-                    let message: OtapArrowBytes = message.try_into()?;
+                Message::PData(pdata) => {
+                    // Capture signal type before moving pdata into try_from
+                    let signal_type = pdata.signal_type();
+
+                    self.pdata_metrics.inc_consumed(signal_type);
```

Comment:
It looks like we are executing quite a bit of `match` expressions here which could be optimized:
- `pdata.signal_type()` would involve two `match` expressions
- `inc_consumed()` would involve an additional `match` expression

We anyway have to match on `message`, so we could simply use the signal specific consumed metric such as `metrics_consumed` instead of calling the helper method `inc_consumed()` which would require three `match` expressions.

Reply:
I made some changes this morning which not related to this comment but the result is much simpler.